### ISSUE TITLE
Appending to table not extending formulas/format

### ISSUE THREAD

**Issue by jfbradfo** [dfinke/ImportExcel#821]
First, thanks so much for this module.  It's a huge time saver!

I have some automation that is appending to an existing table that has a couple of calculated columns.  If I manually add rows by hand or cut/paste, the calculated column automatically fills in, so I know the Excel configuration is correct.  However, when my script does (effectively) the same thing with Export-Excel, and I open the workbook, these columns are blank (I'm importing $null for these columns).  I can manually pull the columns down, but that's not ideal.  I have also tried using the '-calculate' parameter, but that makes no difference.

I have tried this both by passing the path to the workbook and by opening and passing an ExcelPackage object.   Here's my current usage:

`Export-Excel -ExcelPackage $excelPkg -InputObject $eventObject -WorksheetName "$excelWorksheet" -Append -Calculate -Verbose`

I've also tried specifying the table name (-TableName), but it still doesn't work. (In fact, it changed my table format, but that's a different issue.)

This also seems to apply to cells with specific formats, in my case dates.  I know that there is an explicit forma set in Export-Excel, but I've also tried passing a numeric value (days since 12/31/1900), but that does't work either.  Manual entry or copy/paste gives me the correct format (date or date value), but Export-Excel does not.

The column calculations are of two types.  One is a table lookup in another table in the same workbook, and the other is a date calculation from the current table being updated.  

My assumption is that I'm missing something, but this feels like something that's not working right because of the way the cmdlet does the updates (I suspect).

**Comment by dfinke** [comment]
If you could post a small repro, it'd be great.

@jhoneill doesn't look like the table will be appended to. I had assumed it was only the data in a sheet that worked with.

https://github.com/dfinke/ImportExcel/blob/3130020e20e3faaa1cacb4b83dcaf56d2cc99e1b/Public/Export-Excel.ps1#L159

**Comment by jhoneill** [comment]
> If you could post a small repro, it'd be great.
> 
> @jhoneill doesn't look like the table will be appended to. I had assumed it was only the data in a sheet that worked with.
> 
> https://github.com/dfinke/ImportExcel/blob/3130020e20e3faaa1cacb4b83dcaf56d2cc99e1b/Public/Export-Excel.ps1#L159

IIRC if there is an existing table and you do an append it should grow the table to cover the existing rows. But that can't detect tables which don't start at A1 (so a title row would prevent an append from extending the table)

>I have some automation that is appending to an existing table that has a couple of calculated columns.

You will need to have a property with that column name which holds the formula as a string beginning with =   the table extension just changes the bottom right corner of the table.  It won't look at the what was in the data previously ...
Excel does some clever filling in of data when you make an extra table column, we don't even try to reproduce that. 

>apply to cells with specific formats, in my case dates. I know that there is an explicit forma set in Export-Excel, but I've also tried passing a numeric value (days since 12/31/1900), 

Pass a [datetime]  No one knows if 04/05/2020 means April or May, and we don't even try to sort it out. If you put a number in the cell you should be able to apply standard formatting to it.

**Comment by jfbradfo** [comment]
For clarity, this is a single table starting at A1 on a dedicated worksheet, so the first note doesn't apply.  

I am not adding columns, only rows.  My expectation was the Excel would automatically 

- Extend the table to include the appended rows (it does)
- Copy cell styles (eg font, size) to the appended cells, as appropriate (it does, but this may already be there on the whole column)
- Extend data formatting rules (like dates) to the added cells, as appropriate (it does NOT)
- Copy formulas to the new empty cells that should get them (it does NOT)

I've attached a cheap-and-sleazy contrived repro ([TableAppendRepro.zip](https://github.com/dfinke/ImportExcel/files/4530019/TableAppendRepro.zip)) to demonstrate what I'm seeing.  I'm hopeful you'll see the same behavior.  I do have the date format applied to the entire column and the font/size applied to the entire ErrorCode column, which matches my use case. 

Having said that, @jhoneill's response had me reconsider my approach, as I probably shouldn't be relying on Excel, but rather enforcing what I want in the script.  I've updated my code to use functions, and that seems to have fixed that issue. 

Wit respect to the dates, I made a dumb assumption that Export-Excel was converting dates to Excel date values (floats, which Excel does), and that I could apply any format.  What I discovered (to my chagrin) is the format I want to use gives me a float when I import from Excel, but the default format gives me a DateTime.  I will revert to the default format.  

Thanks for the assistance and patience with a newbie!
